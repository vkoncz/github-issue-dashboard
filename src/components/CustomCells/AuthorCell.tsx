import type { ICellRendererParams } from 'ag-grid-community';

import type { Author } from '../../graphql-client/github-client.model';

type AuthorCellParams = Omit<ICellRendererParams, 'value'> & { value?: Author };

export function AuthorCell({ value }: AuthorCellParams) {
  return (
    <span>
      <a href={value?.url} target="_blank" rel="noreferrer">
        {value?.login}
      </a>
    </span>
  );
}
