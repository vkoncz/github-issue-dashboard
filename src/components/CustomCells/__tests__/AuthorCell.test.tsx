import { render, screen } from '@testing-library/react';

import { AuthorCell } from '../AuthorCell';

describe('AuthorCell', () => {
  it('should contains a link', () => {
    const { container } = render(
      // @ts-ignore
      <AuthorCell value={{ login: 'user name', url: 'https://' }} />,
    );

    const link = screen.getByRole('link');

    expect(link).toBeDefined();
    expect(container).toMatchSnapshot();
  });

  it('should handle undefine value without generate link', async () => {
    // @ts-ignore
    const { container } = render(<AuthorCell />);

    const link = screen.queryByRole('link');

    expect(link).toBeDefined();
    expect(container).toMatchSnapshot();
  });
});
