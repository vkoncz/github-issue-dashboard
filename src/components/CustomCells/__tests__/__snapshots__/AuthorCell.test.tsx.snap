// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthorCell should contains a link 1`] = `
<div>
  <span>
    <a
      href="https://"
      rel="noreferrer"
      target="_blank"
    >
      user name
    </a>
  </span>
</div>
`;

exports[`AuthorCell should handle undefine value without generate link 1`] = `
<div>
  <span>
    <a
      rel="noreferrer"
      target="_blank"
    />
  </span>
</div>
`;
